src/Byng/Migration/WebsiteTranslation.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Byng\Pimcore\Migration;
+
+use Pimcore\Model\Translation;
+use Pimcore\Tool;
+use Pimcore\Model\Translation\Website as SiteTranslation;
+
+/**
+ * Description of SharedTranslation
+ *
+ * @author Asim Liaquat <asim@byng.co>
+ */
+class WebsiteTranslation
+{
+    
+    /**
+     * Add new translations
+     * 
+     * The method accepts an array of translations where the key is the
+     * translation key and the value is an array where each row is a translation
+     * for a specific language. e.g.
+     * 
+     *  $translations = [
+     *      "btn_continue" => [
+     *          "en" => "Continue"
+     *      ],
+     *   ];
+     * 
+     * @param array $translations
+     */
+    public function create(array $translations = [])
+    {
+        Translation\Website::clearDependentCache();
+
+        foreach ($translations as $key => $translations) {
+            try {
+                $t = SiteTranslation::getByKey($key);
+            }
+            catch (\Exception $e) {
+
+                $t = new SiteTranslation();
+                $t->setKey($key);
+                $t->setCreationDate(time());
+                $t->setModificationDate(time());
+
+                foreach (Tool::getValidLanguages() as $lang) {
+                    if (isset($translations[$lang])) {
+                        $t->addTranslation($lang, $translations[$lang]);
+                    } else {
+                        $t->addTranslation($lang, "");
+                    }
+                }
+                $t->save();
+            }
+        }
+    }
+    
+    /**
+     * Update translations
+     * 
+     * @param array $translations Same as the create method
+     */
+    public function update(array $translations = [])
+    {
+        Translation\Website::clearDependentCache();
+
+        foreach ($translations as $key => $translations) {
+            try {
+                $t = SiteTranslation::getByKey($key);
+                $t->setModificationDate(time());
+                foreach (Tool::getValidLanguages() as $lang) {
+                    if (isset($translations[$lang])) {
+                        $t->addTranslation($lang, $translations[$lang]);
+                    } else {
+                        $t->addTranslation($lang, "");
+                    }
+                }
+                $t->save();
+            }
+            catch (\Exception $e) {}
+        }
+    }
+    
+    /**
+     * Delete a translation
+     * 
+     * @param string $key
+     */
+    public function delete($key)
+    {
+        $translation = SiteTranslation::getByKey($key);
+        if ($translation) {
+            $translation->delete();
+        }
+    }
+    
+}
```

Comment:
The white page of death issue is caused by having PHP closing tags at the end of files that don't need them, you couldn't ever have the issue if you don't have a closing tag :)

It is pretty standard to have a blank line at the end of your files, for a few reasons, they're quite simple really, the ones that spring to mind instantly are; it's easy to tell that a file hasn't been truncated, and also that if you are using some kind of tools with your code, they are less likely to break because some older tools will require the new line.

The other benefit also being that if you want to go the end of the file, you're not going to be stuck in the middle of a bunch of code somewhere, you'll land on a blank line.